packages/lib-classifier/src/store/DrawingStore.js

```diff
@@ -1,56 +1,156 @@
-import { addDisposer, getRoot, onAction, types } from 'mobx-state-tree'
+import cuid from 'cuid'
+import { addDisposer, getRoot, getType, onAction, types } from 'mobx-state-tree'
 import { autorun } from 'mobx'
 import { Subject } from 'rxjs'
 
+import { Line, Point } from './markings'
+
+const markModels = {
+  line: Line,
+  point: Point
+}
+
+function getMarkModel (toolType) {
+  return markModels[toolType] || null
+}
+
 const DrawingStore = types
   .model('DrawingStore', {
-    activeDrawingTool: types.optional(types.number, 0)
+    activeDrawingToolIndex: types.optional(types.number, 0),
+    activeMark: types.safeReference(types.union({
+      dispatcher: getType
+    }, Line, Point)),
+    marks: types.map(types.union({
+      dispatcher: getType
+    }, Line, Point))
   })
+  .views(self => ({
+    get activeDrawingTask () {
+      const [task] = getRoot(self).workflowSteps.activeStepTasks
+        .filter(task => task.type === 'drawing')
+      return task
+    },
+    get isDrawingInActiveWorkflowStep () {
+      return getRoot(self).workflowSteps.activeStepTasks
+        .some(task => task.type === 'drawing')
+    }
+  }))
   .volatile(self => ({
-    eventStream: new Subject()
+    eventStream: new Subject(),
+    svg: null,
+    createSVGPoint: null,
+    getScreenCTMInverse: null
   }))
   .actions(self => {
     function afterAttach () {
+      createDrawingTaskObserver()
       createClassificationObserver()
       createWorkflowStepsObserver()
     }
 
+    function createDrawingTaskObserver () {
+      const drawingTaskDisposer = autorun(() => {
+        if (self.isDrawingInActiveWorkflowStep) {
+          self.createMark()
+        }
+      }, { name: 'DrawingStore DrawingTask Observer autorun' })
+      addDisposer(self, drawingTaskDisposer)
+    }
+
     function createClassificationObserver () {
       const classificationDisposer = autorun(() => {
-        onAction(getRoot(self).classifications, (call) => {
+        onAction(getRoot(self), (call) => {
           if (call.name === 'completeClassification') self.reset()
         })
-      })
+      }, { name: 'DrawingStore Classification Observer autorun' })
       addDisposer(self, classificationDisposer)
     }
 
     function createWorkflowStepsObserver () {
       const workflowStepsDisposer = autorun(() => {
-        onAction(getRoot(self).workflowSteps, (call) => {
+        onAction(getRoot(self), (call) => {
           if (call.name === 'selectStep') self.reset()
         })
-      })
+      }, { name: 'DrawingStore WorkflowStep Observer autorun' })
 
       addDisposer(self, workflowStepsDisposer)
     }
 
+    function createMark () {
```

Comment:
I think creating a new mark should be the responsibility of the active tool eg. `activeTool.createMark()`. Otherwise, I think the code gets into mess trying to figure out what form a new Mark should take. Each tool 'knows' what form its particular marks should take.

Reply:
Resolved in #1266.